modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/src/main/java/com/liferay/headless/admin/workflow/internal/resource/v1_0/WorkflowDefinitionLinkResourceImpl.java

```diff
@@ -133,24 +137,97 @@ public WorkflowDefinitionLink postWorkflowDefinitionWorkflowDefinitionLink(
+				contextCompany.getCompanyId());
+
+			groupId = group.getGroupId();
+		}
+
+		com.liferay.portal.kernel.model.WorkflowDefinitionLink
+			serviceBuilderWorkflowDefinitionLink =
+				_workflowDefinitionLinkService.
+					fetchWorkflowDefinitionLinkByExternalReferenceCode(
+						externalReferenceCode, groupId);
+
+		if (serviceBuilderWorkflowDefinitionLink != null) {
+			return _toWorkflowDefinitionLink(
+				_workflowDefinitionLinkService.updateWorkflowDefinitionLink(
+					groupId, workflowDefinitionLink.getClassName(),
+					serviceBuilderWorkflowDefinitionLink));
+		}
+
+		return _toWorkflowDefinitionLink(
+			_workflowDefinitionLinkService.addWorkflowDefinitionLink(
+				contextUser.getUserId(), contextCompany.getCompanyId(), groupId,
+				workflowDefinitionLink.getClassName(), 0, 0,
+				workflowDefinitionLink.getWorkflowDefinitionName(),
+				workflowDefinitionLink.getWorkflowDefinitionVersion()));
+	}
+
+	private WorkflowDefinitionLink _buildWorkflowDefinitionLink(
+		com.liferay.portal.kernel.model.WorkflowDefinitionLink
+			serviceBuilderWorkflowDefinitionLink,
+		Group group) {
+
+		long groupId = (group != null) ? group.getGroupId() : 0L;
+		String groupExternalReferenceCode =
+			(group != null) ? group.getExternalReferenceCode() :
+				StringPool.BLANK;
+
+		WorkflowDefinitionLink workflowDefinitionLink =
+			new WorkflowDefinitionLink();
+
+		workflowDefinitionLink.setClassName(
+			serviceBuilderWorkflowDefinitionLink::getClassName);
+		workflowDefinitionLink.setExternalReferenceCode(
+			serviceBuilderWorkflowDefinitionLink::getExternalReferenceCode);
+		workflowDefinitionLink.setGroupId(() -> groupId);
+		workflowDefinitionLink.setGroupExternalReferenceCode(
+			() -> groupExternalReferenceCode);
+		workflowDefinitionLink.setId(
+			serviceBuilderWorkflowDefinitionLink::getWorkflowDefinitionLinkId);
+		workflowDefinitionLink.setWorkflowDefinitionName(
+			serviceBuilderWorkflowDefinitionLink::getWorkflowDefinitionName);
+		workflowDefinitionLink.setWorkflowDefinitionVersion(
+			serviceBuilderWorkflowDefinitionLink::getWorkflowDefinitionVersion);
+
+		return workflowDefinitionLink;
+	}
+
 	private WorkflowDefinitionLink _toWorkflowDefinitionLink(
 			com.liferay.portal.kernel.model.WorkflowDefinitionLink
 				workflowDefinitionLink)
 		throws Exception {
 
-		return new WorkflowDefinitionLink() {
-			{
-				setClassName(workflowDefinitionLink::getClassName);
-				setGroupId(workflowDefinitionLink::getGroupId);
-				setId(workflowDefinitionLink::getWorkflowDefinitionLinkId);
-				setWorkflowDefinitionName(
-					workflowDefinitionLink::getWorkflowDefinitionName);
-				setWorkflowDefinitionVersion(
-					workflowDefinitionLink::getWorkflowDefinitionVersion);
-			}
-		};
+		Group group = null;
+
+		if (workflowDefinitionLink.getGroupId() != 0) {
+			group = _groupService.getGroup(workflowDefinitionLink.getGroupId());
+		}
+
+		return _buildWorkflowDefinitionLink(workflowDefinitionLink, group);
```

Comment:
I think you don't need to create a new method for this, you can use lambda to do the specific logic for some fields. Please see this example  [ObjectEntryDTOConverter#toDTO](https://github.com/liferay/liferay-portal/blob/89153ddcb1bd2d4f46c01430eb7f232bd59d287e/modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java#L188-L261)